### PR TITLE
More sysmem tests

### PR DESCRIPTION
**Pull request by joelsmithTT**
### Issue
https://github.com/tenstorrent/tt-umd/issues/293

### Description
Adds more unit tests for sysmem.  I need a reliable way to ensure it is functioning as intended as work ramps up to enable IOMMU and provide a transition away from huge pages.

### List of the changes
* Adds a second sysmem test for Wormhole that examines more of the address space.
* Adds a Grayskull version of the simple Wormhole sysmem test.
* Some whitespace cleanup

### Testing
Manually tested; code paths are in CI

### API Changes
There are no API changes in this PR.

### CODE REVIEW

**Review by broskoTT — APPROVED** [review]

File: tests/grayskull/test_silicon_driver.cpp [hunk 1]
```diff
@@ -402,3 +402,70 @@ TEST(SiliconDriverGS, MultiThreadedMemBar) { // this tests takes ~5 mins to run
...
+
+    uint64_t base_address = device.get_pcie_base_addr_from_device(mmio_chip_id);
+
+    // Buffer that we will use to read sysmem into, then write sysmem from.
+    std::vector<uint8_t> buffer(test_size_bytes, 0x0);
+
+    // Step 1: Fill sysmem with random bytes.
+    fill_with_random_bytes(sysmem, test_size_bytes);
+
+    // Step 2: Read sysmem into buffer.
```
[the remark this answers is not in the document]
Comment (joelsmithTT) [reply tests/grayskull/test_silicon_driver.cpp:452]: Yes, it's sort of a loopback or reflection type mechanism.

Host MMIO -> inbound PCIe TLB window -> NOC -> PCIe -> system bus.

Much easier to test things like iATU config, hugepage pinning, etc. this way than building RV32 code that uses the NOC API and running that code on a Tensix.

File: tests/wormhole/test_silicon_driver_wh.cpp [hunk 2]
```diff
@@ -19,10 +19,14 @@
 inline void fill_with_random_bytes(uint8_t* data, size_t n)
 {
     static std::random_device rd;
-    static std::mt19937 gen(rd());
-    static std::uniform_int_distribution<uint8_t> dis(0, 255);
+    static std::mt19937_64 gen(rd());
+    uint64_t* data64 = reinterpret_cast<uint64_t*>(data);
```
[the remark this answers is not in the document]
Comment (joelsmithTT) [reply tests/wormhole/test_silicon_driver_wh.cpp:23]: This change was to make it go faster, it turns out filling up a gigabyte of buffer one random u8 at a time is pretty slow.  I'll consolidate the code.

File: tests/grayskull/test_silicon_driver.cpp [hunk 3]
```diff
@@ -402,3 +402,70 @@ TEST(SiliconDriverGS, MultiThreadedMemBar) { // this tests takes ~5 mins to run
...
+    device.start_device(tt_device_params{});  // no special parameters
+
+    // PCIe core is at (x=0, y=4) on Grayskull NOC0.
+    const chip_id_t mmio_chip_id = 0;
+    const size_t PCIE_X = 0;    // NOC0
+    const size_t PCIE_Y = 4;    // NOC0
+    const tt_cxy_pair PCIE_CORE(mmio_chip_id, PCIE_X, PCIE_Y);
+    const size_t test_size_bytes = 0x4000;  // Arbitrarilly chosen, but small size so the test runs quickly.
+
+    // Bad API: how big is the buffer?  How do we know it's big enough?
```
[the remark this answers is not in the document]
Comment (joelsmithTT) [reply tests/grayskull/test_silicon_driver.cpp:438]: https://github.com/tenstorrent/tt-umd/issues/300

File: tests/wormhole/test_silicon_driver_wh.cpp [hunk 4]
```diff
@@ -711,3 +715,75 @@ TEST(SiliconDriverWH, SysmemTestWithPcie) {
...
+        uint64_t hi = (lo + ONE_GIG) - 1;
+
+        if (channel == 3) {
+            // TODO: I thought everything past 0xffff'dddd was registers or
+            // something, but a) I don't know what's actually there, and b)
+            // the unusable range seems to be bigger than that... so
+            // restricting to 0x8'f000'0000.
+            hi &= ~0x0fff'ffffULL;
+        }
+        fmt::print("testing range lo={:#x} hi={:#x}\n", lo, hi);
```
[the remark this answers is not in the document]
Comment (joelsmithTT) [reply tests/wormhole/test_silicon_driver_wh.cpp]: Removed it instead (nothing else in here logs)

**Pull request merged**